build_platform.py

```diff
@@ -380,6 +402,16 @@ def test_examples_in_folder(platform, folderpath):
         if os.path.exists(gen_file_name):
             ColorPrint.print_info("generating")
 
+        # check if boards.local.txt should be copied
```

Comment:
Can you build the paths for `platform_boards_local_txt` and `default_boards_local_txt` variables prior to jumping into this conditional? It'd make the code more self-documenting